.gitlab/container_build/docker_linux.yml

```diff
@@ -4,17 +4,52 @@
   script:
     - TAG_SUFFIX=${TAG_SUFFIX:-}
     - BUILD_ARG=${BUILD_ARG:-}
+    - EXTRA_BUILD_CONTEXT=${ARTIFACTS_BUILD_CONTEXT:+"--build-context artifacts=$ARTIFACTS_BUILD_CONTEXT"}
     - |
       if [[ "$BUCKET_BRANCH" == "nightly" && ( "$IMAGE" =~ "ci/datadog-agent/agent" || "$IMAGE" =~ "ci/datadog-agent/cluster-agent" || "$IMAGE" =~ "ci/datadog-agent/cws-instrumentation"  || "$IMAGE" =~ "ci/datadog-agent/otel-agent" ) ]]; then
         export ECR_RELEASE_SUFFIX="-nightly"
       else
         export ECR_RELEASE_SUFFIX=${CI_COMMIT_TAG+-release}
       fi
+    # Caching setup
+    - DOCKER_CACHE_TARGET="${IMAGE}${TAG_SUFFIX}-${ARCH}:cache"
+    # Don't use caching on nightlies, to allow for regular cache invalidation,
+    # and update the cache on both main and nightly builds
```

Comment:
Question for reviewers: would it also make sense for deploy pipelines to run fully uncached?

Reply:
I guess it does, but maybe with a possibility to override it in case maven is down while we're trying to do a release or similar scearios

Reply:
I just added `--no-cache` for deploy pipelines for now. I think in case of an emergency we could probably just edit the pipeline on a release branch to use the cache, I think failures during releases probably don't happen often enough to add any additional mechanism for now.